frontend/src/shared/helpers.tsx

```diff
@@ -617,35 +662,30 @@ export const CohortSamplesDetailsColumns: ColDef[] = [
   },
 ];
 
+setupEditableSampleFields(SampleDetailsColumns);
+setupEditableSampleFields(CohortSampleDetailsColumns);
+
 export const defaultColDef: ColDef = {
   sortable: true,
   resizable: true,
-  headerComponentParams: readOnlyHeader,
-};
-
-export const defaultEditableColDef: ColDef = {
-  ...defaultColDef,
-  editable: true,
-};
-
-export const defaultReadOnlyColDef: ColDef = {
-  ...defaultColDef,
   editable: false,
+  headerComponentParams: readOnlyHeader,
 };
 
-const protectedFields: string[] = [
-  "cmoSampleName",
-  "igoComplete",
-  "importDate",
-  "primaryId",
-  "cmoSampleIdFields",
-  "libraries",
-  "genePanel",
-  "baitSet",
-  "species",
-  "validationStatus",
-  "validationReport",
-  "revisable",
+const editableSampleFields = [
+  "cmoPatientId",
+  "investigatorSampleId",
+  "sampleType",
+  "preservation",
+  "tumorOrNormal",
+  "sampleClass",
+  "oncotreeCode",
+  "collectionYear",
+  "sampleOrigin",
+  "tissueLocation",
+  "sex",
+  "billed",
+  "costCenter",
```

Comment:
Instead of specifying fields that are non-editable, I switched this to a list of fields that are editable. This should (1) help us avoid accidentally making fields editable when they are to be protected and (2) result in less frontend codes.